src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
-	fi
-
-	# print the sar report to the resources file
-	echo "SAR$FLAGS report" >> $RESOURCES_FILE
-	sar$FLAGS >> $RESOURCES_FILE
-}
-
-# print the output of "apachectl fullstatus" to the resources file
+  local OPTION=$1
+  local LOGFILE=$2
+  local FLAGS=''
+  # check to see if we're going to use any parameters for sar
+  if [[ ${OPTION} == "r" ]]; then
+    FLAGS=" -r"
+  elif [[ ${OPTION} == "q" ]]; then
+    FLAGS=" -q"
+  else
+    FLAGS=""
+  fi
+
+  echo "SAR${FLAGS} report" >> "${LOGFILE}"
+  sar${FLAGS} >> "${LOGFILE}"
+}
+
+# print the output of "apachectl fullstatus" to the specified file
 print_httpd_fullstatus() {
-	echo "Apache Status report" >> $RESOURCES_FILE
-	apachectl fullstatus >> $RESOURCES_FILE
+  local LOGFILE=$1
+  echo "Apache Status report" >> "${LOGFILE}"
+  apachectl fullstatus >> "${LOGFILE}"
 }
 
-# print the output of "netstat -ntulpae" to the netstat file
+# print the output of "netstat -ntulpae" to the specified file
 print_netstat() {
-	echo "Network connections" >> $NETSTAT_FILE
-	netstat $OPTS_NETSTAT >> $NETSTAT_FILE
+  local LOGFILE=$1
+  echo "Network connections" >> "${LOGFILE}"
+  netstat ${OPTS_NETSTAT} >> "${LOGFILE}"
 }
 
-# print the output of "netstat -s" to the netstat file
+# print the output of "netstat -s" to the specified file
 print_netstat_sum() {
-	echo "Network traffic summary" >> $NETSTAT_FILE
-	netstat $OPTS_NETSTAT_SUM >> $NETSTAT_FILE
+  local LOGFILE=$1
+  echo "Network traffic summary" >> "${LOGFILE}"
+  netstat ${OPTS_NETSTAT_SUM} >> "${LOGFILE}"
 }
 
 # print the output of "mysqladmin status" to the mysql file
 print_mysql() {
-        # We only support a single DOTMYDOTCNF for MySQL when running Plesk
-        if [ -f /etc/psa/.psa.shadow ]
-        then
-                echo "MySQL status" >> $MYSQL_FILE
-                mysqladmin -uadmin -p`cat /etc/psa/.psa.shadow` status >> $MYSQL_FILE
-                print_blankline $MYSQL_FILE
-        else
-                for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" )
-                do
-                        if [ -f $cnf ]
-                        then
-                                echo "MySQL $cnf status" >> $MYSQL_FILE
-                                mysqladmin --defaults-file=$cnf status >> $MYSQL_FILE
-                                print_blankline $MYSQL_FILE
-                        else
-                                echo "MySQL $cnf for status does not exist. Please check DOTMYDOTCNF in /etc/recap" >> $MYSQL_FILE
-                                print_blankline $MYSQL_FILE
-                        fi
-                done
-        fi
+  local LOGFILE=$1
+  # We only support a single DOTMYDOTCNF for MySQL when running Plesk
+  local PLESK_FILE="/etc/psa/.psa.shadow"
+  if [[ -r "${PLESK_FILE}" ]]; then
+    echo "MySQL status" >> "${LOGFILE}"
+    mysqladmin -uadmin -p$(cat "${PLESK_FILE}") status >> "${LOGFILE}"
+    print_blankline "${LOGFILE}"
+  else
+    for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" ); do
```

Comment:
You're right, in the context of how you read my comment.  Apologies, I didn't communicate the issue very well.  The potential "whitespace problem" is if you've got spaces in the filepath of the my.cnf file.  If the filepath has whitespace, then you'll handle one filepath as two iterations.  The typical fix for *that* whitespace issue would be to quote it, but since for-loops use whitespace as a separator, you'd break your loop by quoting the "variable".

This code illustrates the problem I'm trying to convey:
```
ahoward@galapagos ~ $ multi_confs="/etc/mysql config.cnf"
ahoward@galapagos ~ $ multi_confs="${multi_confs},/etc/my.cnf"
ahoward@galapagos ~ $ for cnf in $( sed -e 's/,/ /g' <<<"${multi_confs}" ); do
> echo "${cnf}"
> done
/etc/mysql
config.cnf
/etc/my.cnf
ahoward@galapagos ~ $ for cnf in "$( sed -e 's/,/ /g' <<<"${multi_confs}" )"; do echo "${cnf}"; done
/etc/mysql config.cnf /etc/my.cnf
ahoward@galapagos ~ $ 
```
Notice, the first for-loop iterates 3x times, and iterations 1 & 2 are each broken due to the partial filename.  The typical fix for that whitespace issue is the quote the var, but that's not an option in for-loops, as illustrated by the second example above.

That's why I'd recommended using "while read LINE; do ... done < <(tr ',' '\n')", or arrays, or just mandate a single mysql-defaults-file instead of trying to handle multiple.